R/simplify_fn.R

```diff
@@ -1,29 +1,37 @@
-simple_simplify_predictions_fn <- function(object, predictions) {
-  predictions %>%
-    purrr::reduce(rbind) %>%
-    tibble::as_tibble()
+as_tf_prediction <- function(x) {
+  structure(x, class = c("tf_prediction", class(x)))
+}
+
+#' @export
+type_sum.tf_prediction <- function(x) {
+  if (is.numeric(x))
+    paste0(signif(x, digits = 3), collapse = ", ")
+  else
+    paste0(x)
+}
+
+simple_simplify_predictions_fn <- function(results) {
+  results %>%
+    purrr::map(~ purrr::map(.x, list)) %>% 
+    purrr::transpose() %>%
+    purrr::map(~ purrr::map(.x, 
+                            purrr::compose(as_tf_prediction, unlist))
```

Comment:
Looks good to me! But would multiple `map`s here affect the performance?

Reply:
Good point! We can probably avoid the `list()`-`unlist()` here; lemme give it a shot.